addons/pos_restaurant/models/pos_config.py

```diff
@@ -63,3 +63,13 @@ def write(self, vals):
         if ('is_order_printer' in vals and vals['is_order_printer'] == False):
             vals['printer_ids'] = [(5, 0, 0)]
         return super(PosConfig, self).write(vals)
+
+    def get_all_table_draft_orders(self):
+        self.ensure_one()
+        self = self.with_context(prefetch_fields=False)
+        order_obj = self.env['pos.order']
+        tables = self.env['restaurant.table'].search([('floor_id.pos_config_id', '=', self.id)])
+        return {
+            table.id: order_obj.get_table_draft_orders(table.id)
+            for table in tables
```

Comment:
The method `get_table_draft_orders` does not work on batches.
So the actual gain of this is to replace N RPC calls by one call that does the N former calls locally, right?

Reply:
you right!

we made it this way because we can't leave aside the get_table_draft_orders, since it's the one with all order fields and it's the one you inherit to extend it.

The proper solution should be to use a read_group, one call, one rpc, that could be proposed to master

cc @fernandahf

Reply:
It's already done starting 16 https://github.com/odoo/odoo/blob/16.0/addons/pos_restaurant/models/pos_order.py#L164-L176